### PR TITLE
Dice

### PR DESCRIPTION
**Pull request by m4rkyma**
Dice.vy Calculate Payout Method, 3 tests working.

### CODE REVIEW

File: contracts/Dice.vy [hunk 1]
```diff
@@ -0,0 +1,74 @@
+from vyper.interfaces import ERC20
+
+# Define constants and types
+MIN_SLIDER: constant(int128) = 1
+MAX_SLIDER: constant(int128) = 100
+
+interface RandomNumberInterface:
+    def get_random_number() -> int128: view
```
Comment (HarvardWestlake) [inline contracts/Dice.vy:8]: We should probably call out in a README or whomever is writing this that they need to link this code.

File: contracts/Dice.vy [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+    """
+    Calculate the percent chance of winning based on sliderInt and rollAbove.
+    """
+    assert MIN_SLIDER <= sliderInt and sliderInt <= MAX_SLIDER, "Slider out of range"
+    if rollAbove:
+        return MAX_SLIDER - sliderInt
+    else:
+        return sliderInt - MIN_SLIDER
+
+<<<<<<< HEAD
```
Comment (HarvardWestlake) [inline contracts/Dice.vy:41]: Merge request <<<<<< is invalid code and should never be in a pull request  👎

**Review by HarvardWestlake — CHANGES_REQUESTED** [review]
theiss@FS-JBJPCY3:~/Blocade/contracts$ vyper Dice.vy
  File "<unknown>", line 41
    <<<<<<< HEAD
    ^^
SyntaxError: invalid syntax

The above exception was the direct cause of the following exception:

vyper.exceptions.SyntaxException: invalid syntax (<unknown>, line 41)
  line 41:1
       40
  ---> 41 <<<<<<< HEAD
  ---------^
       42 @internal

theiss@FS-JBJPCY3:~/Blocade/contracts$

**Review by andrewtheiss — CHANGES_REQUESTED** [review]
Compiles but fails ape test and direct pytest

File compiles but cannot run test

(ape) theiss@Theiss:~/Documents/Github/Blocade$ pytest ./tests/test_Dice.py
INFO: Compiling 'Dice.vy'.
/home/theiss/ape/lib/python3.10/site-packages/_pytest/config/__init__.py:331: PluggyTeardownRaisedWarning: A plugin raised an exception during an old-style hookwrapper teardown.
Plugin: helpconfig, Hook: pytest_cmdline_parse
UsageError: Unable to load project. Use `-v DEBUG` to see more info.
Failure reason: (VyperCompileError) Dice.vy
FileNotFoundError:could not find Dice.vy in any of the following locations:
  /home/theiss/Documents/Github/Blocade/contracts/Dice.vy
For more information see https://pluggy.readthedocs.io/en/stable/api_reference.html#pluggy.PluggyTeardownRaisedWarning
  config = pluginmanager.hook.pytest_cmdline_parse(
ERROR: Unable to load project. Use `-v DEBUG` to see more info.
Failure reason: (VyperCompileError) Dice.vy
FileNotFoundError:could not find Dice.vy in any of the following locations:
  /home/theiss/Documents/Github/Blocade/contracts/Dice.vy

(ape) theiss@Theiss:~/Documents/Github/Blocade$

**Review by andrewtheiss — APPROVED** [review]
Figured out how to get it working correctly.  Added vyper compile details on top of file:

INFO: Compiling 'RNG/RNGV3ExampleGame.vy'.
====================================================================== test session starts ======================================================================
platform linux -- Python 3.10.12, pytest-7.4.4, pluggy-1.5.0
rootdir: /home/theiss/Documents/Github/Blocade
plugins: eth-ape-0.7.18, web3-6.17.2
collected 24 items

tests/test_BattleWyrmToken.py .                                                                                                                           [  4%]
tests/test_Dice.py .....                                                                                                                                  [ 25%]
tests/test_ERC721.py ...                                                                                                                                  [ 37%]
tests/test_WyrmAutoBattler.py .                                                                                                                           [ 41%]
tests/test_aceArcade.py .                                                                                                                                 [ 45%]
tests/test_exampleGame.py ....                                                                                                                            [ 62%]
tests/test_mintToken.py .                                                                                                                                 [ 66%]
tests/test_rockPS.py ..                                                                                                                                   [ 75%]
tests/test_signupTester.py .                                                                                                                              [ 79%]
tests/test_wheeler.py ...                                                                                                                                 [ 91%]
tests/RNG/test_RNGV3.py .                                                                                                                                 [ 95%]
tests/RNG/test_RNGV3exampleGame.py .                                                                                                                      [100%]

====================================================================== 24 passed in 9.53s =======================================================================
(ape) theiss@Theiss:~/Documents/Github/Blocade$

**Review by HarvardWestlake — APPROVED** [review]
+1

**Pull request merged**